sdk/core/azure-core/azure/core/rest/_aiohttp.py

```diff
@@ -0,0 +1,145 @@
+
+
+class RestAioHttpTransportResponse(AsyncHttpResponse):
+    def __init__(
+        self,
+        *,
+        request: HttpRequest,
+        internal_response,
+    ):
+        super().__init__(request=request, internal_response=internal_response)
+        self.status_code = internal_response.status
+        self.headers = CIMultiDict(internal_response.headers)  # type: ignore
+        self.reason = internal_response.reason
+        self.content_type = internal_response.headers.get('content-type')
+        self._decompress = None
+        self._decompressed_content = None
+
+    @property
+    def content(self) -> bytes:
+        if self._content is None:
+            raise ResponseNotReadError()
+        if not self._decompress:
+            return self._content
+        enc = self.headers.get('Content-Encoding')
+        if not enc:
+            return self._content
+        enc = enc.lower()
+        if enc in ("gzip", "deflate"):
+            if self._decompressed_content:
+                return self._decompressed_content
+            import zlib
+            zlib_mode = 16 + zlib.MAX_WBITS if enc == "gzip" else zlib.MAX_WBITS
+            decompressor = zlib.decompressobj(wbits=zlib_mode)
+            self._decompressed_content = decompressor.decompress(self._content)
+            return self._decompressed_content
+        return self._content
+
+    @property
+    def text(self) -> str:
+        content = self.content
+        encoding = self.encoding
+        ctype = self.headers.get(aiohttp.hdrs.CONTENT_TYPE, "").lower()
+        mimetype = aiohttp.helpers.parse_mimetype(ctype)
+
+        encoding = mimetype.parameters.get("charset")
+        if encoding:
+            try:
+                codecs.lookup(encoding)
+            except LookupError:
+                encoding = None
+        if not encoding:
+            if mimetype.type == "application" and (
+                    mimetype.subtype == "json" or mimetype.subtype == "rdap"
+            ):
+                # RFC 7159 states that the default encoding is UTF-8.
+                # RFC 7483 defines application/rdap+json
+                encoding = "utf-8"
+            elif content is None:
+                raise RuntimeError(
+                    "Cannot guess the encoding of a not yet read content"
+                )
+            else:
+                encoding = chardet.detect(content)["encoding"]
+        if not encoding:
+            encoding = "utf-8-sig"
+
+        return content.decode(encoding)
+
+    async def iter_raw(self) -> AsyncIterator[bytes]:
+        """Asynchronously iterates over the response's bytes. Will not decompress in the process
+        :return: An async iterator of bytes from the response
+        :rtype: AsyncIterator[bytes]
+        """
+        async for part in iter_raw_helper(AioHttpStreamDownloadGenerator, self):
+            yield part
+        await self.close()
+
+    async def iter_bytes(self) -> AsyncIterator[bytes]:
```

Comment:
add `type: ignore` here, so I don't have to have gross code to satiate mypy in base class

Reply:
open an issue, link it